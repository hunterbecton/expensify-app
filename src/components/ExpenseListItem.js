import React from 'react';
import { Link } from 'react-router-dom';


const ExpenseListItem = ({description, amount, createdAt, id, dispatch}) => (
    <div>
        <Link to={`/edit/${id}`}><h3>{ description }</h3></Link>
        <p>{ amount } - { createdAt }</p>
    </div>
);

//Nothing is needed form the state, which is why we don't pass anything into the connect function
export default ExpenseListItem;