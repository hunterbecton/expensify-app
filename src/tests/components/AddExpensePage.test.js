import React from 'react';
import { shallow } from 'enzyme';
import { AddExpensePage } from '../../components/AddExpensePage';
import expenses from '../fixtures/expenses';

let onSubmit, history, wrapper;

beforeEach(() => {
    onSubmit = jest.fn();
    history = { push: jest.fn() };
    wrapper = shallow(<AddExpensePage onSubimt={onSubmit} history={history}/>);
});

test('Should render add expense page correctly',() => {
    expect(wrapper).toMatchSnapshot();
});

test('Should handle on submit',() => {
    wrapper.find('ExpenseForm').prop('onSubmit')(expenses[1]);
    expect(history.push).toHaveBeenLastCalledWith('/');
    expect(onSubmit).toHaveBeenLastCalledWith(expenses[1]);
});