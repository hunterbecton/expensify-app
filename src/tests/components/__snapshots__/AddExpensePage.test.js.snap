// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render add expense page correctly 1`] = `""`;
